recommender/services/fts.py

```diff
@@ -44,6 +45,33 @@ def filter_services(search_data: SearchData) -> QuerySet:
 
 
 def retrieve_services_for_recommendation(
+    elastic_services: List[int], accessed_services: Optional[Iterable] = None
+) -> QuerySet:
+    """
+    Selecting candidates for recommendation
+
+    Args:
+        elastic_services: Marketplace's context
+        accessed_services: Services that user accessed
+    """
+
+    q = deepcopy(elastic_services)
+    q = check_services_availability_and_status(q)
+    q = q(id__nin=[s.id for s in accessed_services]) if accessed_services else q
+    return q
+
+
+def check_services_availability_and_status(q: List[int]) -> QuerySet:
```

Comment:
better name: `filter_available_and__recommendable_services` because this function not only check but also modify input and returns such changed output